### ISSUE TITLE
CA2012 Bug: Triggers when ValueTask direct return includes a property

### ISSUE THREAD

**Issue by NickCraver** [dotnet/roslyn-analyzers#4271]
### Analyzer

**Diagnostic ID**: [CA2012](https://docs.microsoft.com/visualstudio/code-quality/ca2012): `Use ValueTasks correctly`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100-rc.1.20452.10](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

This was an oddity @stephentoub and I came across yesterday, it looks like specifically referencing a property in a `ValueTask` return triggers the warning where any other combination doesn't.

### Steps To Reproduce

Minimal Repro case:
```cs
class Program
{
    string Role { get; set; }
    static void Main(string[] args) { }
    ValueTask<(string, string)> Foo0Async() => ValueTask.FromResult(("", Role));  // CA2012 Triggers
    ValueTask<(string, string)> Foo1Async() => ValueTask.FromResult(("", ""));    // Fine
    ValueTask<(string, string)> Foo2Async() => ValueTask.FromResult(("", Bar())); // Fine
    string Bar() => "";
}
```

### Expected behavior
No warning is triggered, code is as intended (a `ValueTask` is returned for awaiting elsewhere).

### Actual behavior
Analyzer triggers a warning:
> Program.cs(12,20): error CA2012: ValueTask instances returned from method calls should be directly awaited, returned, or passed as an argument to another method call. Other usage, such as storing an instance into a local or a field, is likely an indication of a bug, as ValueTask instances must only ever be consumed once. [/Users/ncraver/git/NickCraver/CA2012Repro/CA2012Repro.csproj]

## Additional context

I've tried to include examples of the inconsistency above, showing other approaches that _don't_ error very similarly. In our entire solution, this was the only odd-ball from turning CA2012 on - lots of similar cases didn't trigger...there's something specific to the property on this one.

**Comment by Youssef1313** [comment]
For some reason GitHub didn't auto-close the issue when #4274 is merged.

**Comment by stephentoub** [comment]
Thanks
